NEWS.md

```diff
@@ -96,6 +96,8 @@
 
 14. `setNumericRounding()` now invisibly returns the old rounding value instead of `NULL`, which is now consistent with similar behavior by `setwd()`, `options()`, etc. Thanks @MichaelChirico for the report and @joshhwuu for the fix.
 
+15. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()` functions. Previously, only `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` were exported. This change helps package developers avoid defining these constructs inside functions, thereby preventing CRAN NOTES and ensuring consistency in data.table usage. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing this improvement.
```

Comment:
again, what is consistency? please either delete or clarify

Reply:
consistency here means that  the behaviour of `.`, `J`, `measure`, and `patterns` constructs after being exported is predictable and aligning with the previously exported constructs like .N, .I, .GRP, .GRPI, .SD, and :=, like before we had to define them inside their functions but didn't have to do that with others which were exported, so after this PR,  behaviour of all NSE constructs are consistent,  i think #5604 had the consistency tag for same reason.

Reply:
great that makes sense, please edit the NEWS item to clarify that meaning of consistency.
how about "... are now exported for use withing [ and melt functions, for consistency with `.N` ... which were the only previous NSE exports."

Reply:
updated 👍